raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -252,6 +252,13 @@ def clear_if_finalized(iteration, channelidentifiers_to_channels):
         if channel.is_lock_pending(payee_channel.our_state, secrethash):
             return iteration
 
+    if state.waiting_transfer:
```

Comment:
perhaps add a link to the issue here? It would be a nice thing to refer to why we have this branch in `clear_if_finalized`

Reply:
I would prefer avoiding relying on github's numbering, if you want I can do it. (this is explained by the tests, btw)

Reply:
We already do it in many places in the code.

 I personally find it very useful to check the issue that lead to the code like this and read more on our thinking at the time and the decisions that were made which lead to the code/architecture.

Won't keep the PR for this ofcourse. In the end this is just a nitpick.